### ISSUE TITLE
GTK: Add an option to hide window decorations when the window is maximized

### ISSUE THREAD

**Issue by ansipunk** [ghostty-org/ghostty#3381]
I find window decorations useless when the window is maximized, so I'd like to hide them to preserve screen space, since I use tmux for multiplexing the terminal. Foot has an option for that and I imagine it wouldn't be too difficult to implement it here.

**Comment by ohmsl** [comment]
Could use
https://ghostty.org/docs/config/reference#window-decoration
if you want to always disable it

**Comment by ansipunk** [comment]
@ohmsl then I'd lose the ability to pull edges and corners of the window, move it around when it's not maximized and would have to use a keybinding to toggle. I think it could be improved.

**Comment by ohmsl** [comment]
Agreed, it should at least remove the window decorations when full screened, whether this needs to be an option I don't know but +1 from me

**Comment by mitchellh** [comment]
I prefer a discussion for feature requests but I'm willing to accept this one on the spot. It's a good idea. 

This should only apply to GTK. macOS already has its own hidden decorations when fullscreen (that appear when you hover the mouse up top).

**Comment by isgj** [comment]
IMO it should hide the decorations when it goes full-screen,  not when the window is maximized.

**Comment by Pangoraw** [comment]
@ansipunk would something like this help ?

**Comment by ansipunk** [comment]
@Pangoraw thanks, I'll keep on eye on that discussion, it indeed does look great.

**Comment by yorickpeterse** [comment]
The trick I was using was to use `ghostty --gtk-titlebar=false` as this would result in a basic GTK titlebar, which _did_ get hidden when enabling fullscreen mode. It seems that per 1.0.1 this approach no longer works, as `--gtk-titlebar=false` completely removes the title bar on Gnome.

In my case I use Neovim in Ghostty in fullscreen mode, and not being able to hide the title bar in full screen mode is rather annoying due to the amount of space it takes up.

**Comment by Doom4535** [comment]
I might add that an auto hide type behavior would be nice (like how the Windows taskbar can be configured to disappear when the mouse isn't near it), as the GUI buttons are convenient, but for smaller (and lower res) screens real-estate is at a premium and the decorations bar takes up a good amount of space (or at least on my Ubuntu system).  I am using the `window-decoration = false` setting currently to work around this, but sometimes miss the buttons.

**Comment by ansipunk** [comment]
Please do not confuse full screen and maximized modes.

**Comment by jcollie** [comment]
I'd say that the titlebar and windows should be automatically hidden when the window goes into fullscreen, not maximized. For maximized the decorations and titlebar would not change, only the window would be bigger.

**Comment by ansipunk** [comment]
@jcollie this issue was created specifically for hiding decorations in maximized windows. For full screen windows create another one.

**Comment by ak1932** [comment]
I think a better option would be to have auto hiding titlebar. The terminal Blackbox handles this pretty well. It hides the titlebar and when you move the cursor near the top it shows the titlebar. This way you don't have clutter as well as have resizable windows from corners, and a way to use the tab management ui that ghostty has.

**Comment by sergionexus** [comment]
> [@jcollie](https://github.com/jcollie) this issue was created specifically for hiding decorations in maximized windows. For full screen windows create another one.

I create another one and a subnormal deleted it. another good idea?

**Comment by ansipunk** [comment]
@jcollie are you fucking with me? This issue is maximized specific and has its goals defined with a config option that is not enabled by default.

**Comment by pluiedev** [comment]
Please have basic courtesy when interacting with others in the issue tracker.

**Comment by mitchellh** [comment]
Please remain civil and kind @ansipunk. 

I still think this is a good idea _as an option_. And I understand its for maximized windows. I can understand the reasoning for wanting this and I also think it's a good sign that other programs such as Foot offer this.
